x/ecocredit/server/msg_server.go

```diff
@@ -777,12 +777,24 @@ func (s serverImpl) Buy(goCtx context.Context, req *ecocredit.MsgBuy) (*ecocredi
 				return nil, err
 			}
 
+			// error if auto-retire is required for given sell order
+			if !sellOrder.DisableAutoRetire && order.DisableAutoRetire {
```

Comment:
i think it would be nice to add another quick check:

```go
if order.RetirementLocation == "" && sellOrder.DisableAutoRetire {
    return someError
}
```

Reply:
Good catch but retirement location is optional.

```
    // retirement_location is the optional retirement location for the credits
    // which will be used only if disable_auto_retire is false.
```

Reply:
but if sellOrder has auto retire enabled, then the retirement location would be required

Reply:
I might have misunderstood. I was reading this as retirement location is always optional. If sell order has auto-retire enabled, the retirement location will be stored either as an empty string or a valid location. An empty location is not valid when executing `MsgRetire` so I think you are right. I'll add a check, thanks!